### PR TITLE
Modular pack design

### PR DESCRIPTION
**Pull request by ksatirli**
Hey @workloads/reviewers-nomad,

This PR adds the first steps for modularizing _all_ Packs and removing duplicate code by pulling it from a `utility` Pack via helper templates.

### CODE REVIEW

File: packs/utilities/templates/_group/volume.tpl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[[/* iterate over `var.volumes` to create Volumes */]]
```
Comment (sofixa) [inline packs/utilities/templates/_group/volume.tpl:1]: What about the `volume_mount` block?

File: packs/utilities/metadata.hcl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+app {
+  url = "https://github.com/workloads/nomad-pack-registry"
+}
+
+pack {
+  name        = "utilities"
+  description = "A Nomad Pack with common utilities"
+  url         = "https://github.com/workloads/nomad-pack-registry/tree/main/packs/utilities"
+  version     = "0.0.0"
```
Comment (sofixa) [inline packs/utilities/metadata.hcl:9]:
```suggestion
  version     = "0.0.1"
```

File: packs/utilities/metadata.hcl [hunk 3]
```diff
@@ -0,0 +1,10 @@
+app {
+  url = "https://github.com/workloads/nomad-pack-registry"
+}
+
+pack {
+  name        = "utilities"
+  description = "A Nomad Pack with common utilities"
```
Comment (sofixa) [inline packs/utilities/metadata.hcl:7]:
```suggestion
  description = "A Nomad Pack with common utilities. Not to be used directly but to be included as a dependency in other Packs."
```

File: packs/utilities/templates/_group/volume.tpl (hunk 1 above)
Comment (ksatirli) [reply packs/utilities/templates/_group/volume.tpl:1]: Good one, added a `volume_mount` template helper

**Pull request merged**